gtsam_integrator/plugin.cpp

```diff
@@ -221,11 +232,8 @@ class imu_integrator : public plugin {
         std::cout << "Integrating over " << prop_data.size() << " IMU samples\n";
 #endif
 
-        for (std::size_t i = 0; i < prop_data.size() - 1; i++) {
-            const imu_type& prop_datum_i      = prop_data[i];
-            const imu_type& prop_datum_i_next = prop_data[i + 1];
-
-            _pim_obj->integrateMeasurement(prop_datum_i, prop_datum_i_next);
+        for (int i = 0; i < int(prop_data.size()) - 1; i++) {
```

Comment:
Not sure what happened to Gio's comment here. @e3m3 what did you mean by the redundant bounds checking for this for loop? Which previous checks are you talking about?

Reply:
> Not sure what happened to Gio's comment here. @e3m3 what did you mean by the redundant bounds checking for this for loop? Which previous checks are you talking about?

It disappeared because I think @charmoniumQ resolved it. Answering in more detail for clarity:

> Returns a reference to the element at specified location pos, with bounds checking.

From [cppreference](https://en.cppreference.com/w/cpp/container/vector/at).

> ... They are mainly useful for catching bugs in your code. If you need to bounds-check at runtime because e.g. the index comes from user input, you're indeed best off with an if statement. ...

From this [SO post](https://stackoverflow.com/questions/9376049/vectorat-vs-vectoroperator).

*(From `gtsam_integrator/plugin.cpp`)*
```cpp
/// Need to integrate over a sliding window of 2 imu_type values.
/// If the container of data is smaller than 2 elements, return early.
if (prop_data.size() < 2) {
        return;
}
ImuBias prev_bias = _pim_obj->biasHat();
ImuBias bias = _pim_obj->biasHat();

#ifndef NDEBUG
std::cout << "Integrating over " << prop_data.size() << " IMU samples\n";
#endif

for (int i = 0; i < int(prop_data.size()) - 1; i++) {
```
As for the bound checks, the size of `prop_data` is checked at the `for` and the `if` above. Alse, see the comment regarding the integration window.